tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,189 @@
+#include "OMPLTestHelpers.hpp"
+#include "../../constraint/MockConstraints.hpp"
+#include <aikido/planner/ompl/Planner.hpp>
+#include <aikido/planner/ompl/CRRT.hpp>
+#include <aikido/planner/ompl/CRRTConnect.hpp>
+#include <aikido/trajectory/Interpolated.hpp>
+#include <aikido/constraint/uniform/RnBoxConstraint.hpp>
+#include <aikido/constraint/CartesianProductSampleable.hpp>
+#include <aikido/constraint/CartesianProductTestable.hpp>
+#include <aikido/constraint/JointStateSpaceHelpers.hpp>
+#include <aikido/util/StepSequence.hpp>
+#include <aikido/planner/ompl/MotionValidator.hpp>
+#include <ompl/geometric/planners/rrt/RRTConnect.h>
+
+using StateSpace = aikido::statespace::dart::MetaSkeletonStateSpace;
+using aikido::planner::ompl::getSpaceInformation;
+using aikido::planner::ompl::CRRT;
+using aikido::planner::ompl::CRRTConnect;
+
+namespace {
+
+  template <typename T>
+  double computeTrajLength(T& traj,
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+      aikido::distance::DistanceMetricPtr& dmetric)
+  {
+    auto stateCurrent = stateSpace->createState();
+    auto stateNext = stateSpace->createState();
+
+    double trajDistance = 0.0;
+    
+    for(size_t i = 0; i < traj->getDuration(); ++i)
+    {
+      traj->evaluate(i, stateCurrent);   
+      traj->evaluate(i+1, stateNext);
```

Comment:
This evaluates the trajectory past its end. The simplest fix is to change the loop condition to`getNumWaypoints() - 1`.  If you do that, then you need to handle the special case where `getNumWaypoints() == 0` to prevent overflow from `-1` to `UNIT_MAX`.

Another option is to initialize `i = 1` and to change the indicies in the loop to `i - 1` and `i`.